packages/babel-parser/src/plugins/typescript/scope.ts

```diff
@@ -143,14 +143,16 @@ export default class TypeScriptScopeHandler extends ScopeHandler<TypeScriptScope
   }
 
   checkLocalExport(id: N.Identifier) {
-    const topLevelScope = this.scopeStack[0];
     const { name } = id;
-    if (
-      !topLevelScope.types.has(name) &&
-      !topLevelScope.exportOnlyBindings.has(name) &&
-      !this.hasImport(name)
-    ) {
-      super.checkLocalExport(id);
+
+    if (this.hasImport(name)) return;
+
+    const len = this.scopeStack.length;
+    for (let i = len - 1; i >= 0; i--) {
+      const stack = this.scopeStack[i];
+      if (stack.types.has(name) || stack.exportOnlyBindings.has(name)) return;
```

Comment:
This is a stack of scopes, not a scope of stacks :stuck_out_tongue: 
```suggestion
      const scope = this.scopeStack[i];
      if (scope.types.has(name) || scope.exportOnlyBindings.has(name)) return;
```

Reply:
Wow, I didn't notice this lol. :)